starknet/src/tests/test_stark_tx_auth.cairo

```diff
@@ -0,0 +1,313 @@
+#[cfg(test)]
+mod tests {
+    use array::ArrayTrait;
+    use starknet::{
+        ContractAddress, syscalls::deploy_syscall, testing, contract_address_const, info
+    };
+    use traits::{Into, TryInto};
+    use result::ResultTrait;
+    use option::OptionTrait;
+    use integer::u256_from_felt252;
+    use clone::Clone;
+    use serde::{Serde};
+    use sx::space::space::{Space, ISpaceDispatcher, ISpaceDispatcherTrait};
+    use sx::authenticators::stark_tx::{
+        StarkTxAuthenticator, IStarkTxAuthenticatorDispatcher, IStarkTxAuthenticatorDispatcherTrait
+    };
+    use sx::execution_strategies::vanilla::VanillaExecutionStrategy;
+    use sx::voting_strategies::vanilla::VanillaVotingStrategy;
+    use sx::proposal_validation_strategies::vanilla::VanillaProposalValidationStrategy;
+    use sx::tests::setup::setup::setup::{setup, deploy};
+    use sx::types::{
+        UserAddress, Strategy, IndexedStrategy, Choice, FinalizationStatus, Proposal,
+        UpdateSettingsCalldataImpl, UpdateSettingsCalldata
+    };
+    use sx::tests::utils::strategy_trait::{StrategyImpl};
+
+    fn setup_stark_tx_auth(
+        space: ISpaceDispatcher, owner: ContractAddress
+    ) -> IStarkTxAuthenticatorDispatcher {
+        // Deploy Stark Tx Authenticator and whitelist in Space
+        let (stark_tx_authenticator_address, _) = deploy_syscall(
+            StarkTxAuthenticator::TEST_CLASS_HASH.try_into().unwrap(), 0, array![].span(), false
+        )
+            .unwrap();
+
+        let authenticator = IStarkTxAuthenticatorDispatcher {
+            contract_address: stark_tx_authenticator_address, 
+        };
+        let mut updateSettingsCalldata: UpdateSettingsCalldata =
+            UpdateSettingsCalldataImpl::default();
+        updateSettingsCalldata.authenticators_to_add = array![stark_tx_authenticator_address];
+        testing::set_contract_address(owner);
+        space.update_settings(updateSettingsCalldata);
+
+        authenticator
+    }
+
+    #[test]
+    #[available_gas(10000000000)]
+    fn test__propose_update_vote() {
+        let config = setup();
+        let (factory, space) = deploy(@config);
+        let authenticator = setup_stark_tx_auth(space, config.owner);
+
+        let quorum = u256_from_felt252(1);
+        let mut constructor_calldata = ArrayTrait::<felt252>::new();
```

Comment:
can we use `let mut constructor_calldata: Array<felt252> = array![]` here?

Reply:
yep

Reply:
im not sure how to do this while serializing the types into the array of felts. any ideas?

Reply:
I don't think you can do this in a one liner but here I was just referring to using `array![]` instead of `ArrayTrait::<felt252>::new()`

Reply:
serializing will still need to be done on the line just after i'm afraid